assets/training/model_management/src/azureml/model/mgmt/utils/common_utils.py

```diff
@@ -275,10 +275,8 @@ def newfn(*args, **kwargs):
 def fetch_huggingface_model_info(model_id) -> ModelInfo:
     """Return Hugging face model info."""
     try:
-        model_list: List[ModelInfo] = hf_api.list_models(filter=ModelFilter(model_name=model_id))
-        for info in model_list:
-            if model_id == info.modelId:
-                return info
+        exists = hf_api.repo_exists(repo_id=model_id)
+        return hf_api.model_info(model_id) if exists else None
```

Comment:
(nit)

This code change is correct and preserves the current behavior of `fetch_huggingface_model_info` (while been more robust/future-proof).

However, the method signature suggests that the return type should be `ModelInfo`, not `Optional[ModelInfo]`. So the implementation could be:

```suggestion
        return hf_api.model_info(model_id)
```

which will raise an error instead of returning None in case the repo does not exists.

Note: this is kinda orthogonal to this PR. Current behavior is also incorrectly returning None if model does not exists.